datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java

```diff
@@ -145,29 +145,42 @@ public SdxRestoreResponse triggerDatalakeRestore(SdxCluster sdxCluster, String d
-            datalakeDRProto.DatalakeBackupInfo backupInfo = datalakeDrClient.getBackupById(datalakeName, backupId, userCrn);
+            datalakeDRProto.DatalakeBackupInfo backupInfo = datalakeDrClient.getBackupById(datalakeName, backupId, userCrn, isMockCloudProvider(sdxCluster));
             if (backupInfo == null) {
                 LOGGER.error("Backup Id {} does not exist for data lake: {}", backupId, datalakeName);
                 throw new NotFoundException(String.format("Backup Id %s does not exist for data lake: %s", backupId, datalakeName));
             }
             backupLocation  = backupInfo.getBackupLocation();
         } else {
-            datalakeDRProto.DatalakeBackupInfo lastSuccessBackupInfo = getLastSuccessfulBackupInfo(datalakeName, userCrn);
+            datalakeDRProto.DatalakeBackupInfo lastSuccessBackupInfo = getLastSuccessfulBackupInfo(sdxCluster, userCrn);
             backupId = lastSuccessBackupInfo.getBackupId();
             backupLocation = lastSuccessBackupInfo.getBackupLocation();
         }
         return triggerDatalakeRestoreFlow(sdxCluster, backupId, backupLocation, backupLocationOverride);
     }
 
-    public datalakeDRProto.DatalakeBackupInfo getLastSuccessfulBackupInfo(String datalakeName, String userCrn) {
-        datalakeDRProto.DatalakeBackupInfo lastSuccessfulBackupInfo = datalakeDrClient.getLastSuccessfulBackup(datalakeName, userCrn, Optional.empty());
+    public datalakeDRProto.DatalakeBackupInfo getLastSuccessfulBackupInfo(SdxCluster sdxCluster, String userCrn) {
+        datalakeDRProto.DatalakeBackupInfo lastSuccessfulBackupInfo = datalakeDrClient.getLastSuccessfulBackup(
+                sdxCluster.getClusterName(), userCrn, Optional.empty(), isMockCloudProvider(sdxCluster)
+        );
         if (lastSuccessfulBackupInfo == null) {
-            LOGGER.error("No successful backup found for data lake: {}", datalakeName);
-            throw new NotFoundException(String.format("No successful backup found for data lake: %s", datalakeName));
+            LOGGER.error("No successful backup found for data lake: {}", sdxCluster.getClusterName());
+            throw new NotFoundException(String.format("No successful backup found for data lake: %s", sdxCluster.getClusterName()));
         }
         return lastSuccessfulBackupInfo;
     }
 
+    public Optional<datalakeDRProto.DatalakeBackupInfo> getLastSuccessfulBackupInfoWithRuntime(SdxCluster sdxCluster, String userCrn, String runtime) {
+        datalakeDRProto.DatalakeBackupInfo lastSuccessfulBackupInfo = datalakeDrClient.getLastSuccessfulBackup(
+                sdxCluster.getClusterName(), userCrn, Optional.of(runtime), isMockCloudProvider(sdxCluster)
+        );
+        if (lastSuccessfulBackupInfo == null) {
+            LOGGER.error("No successful backup found for data lake: {}", sdxCluster.getClusterName());
+            return Optional.empty();
+        }
+        return Optional.of(lastSuccessfulBackupInfo);
+    }
+
```

Comment:
Can you elaborate on this change? Looks like a non-related change.

Reply:
See my comment on your other comment.